test/optimus/pr/ValidateOptimus.wdl

```diff
@@ -1,64 +1,223 @@
-task ValidateOptimus {
-      File bam
-      File matrix
-      File gene_metrics
-      File cell_metrics
+workflow ValidateOptimus {
+     meta {
+         description: "Validate Optimus Outputs"
+     }
 
-      Int required_disk = ceil((size(bam, "G") + size(matrix, "G")) * 1.1)
+     File bam
+     File matrix
+     File matrix_row_index
+     File matrix_col_index
+     File cell_metrics
+     File gene_metrics
 
-      String expected_bam_hash
-      String expected_matrix_hash
-      String expected_gene_metric_hash
-      String expected_cell_metric_hash
+     File reference_matrix
+     String expected_bam_hash
+     String expected_cell_metric_hash
+     String expected_gene_metric_hash
 
-  command <<<
+     call ValidateBam as ValidateBam {
+         input:
+            bam = bam,
+            expected_checksum = expected_bam_hash
+     }
 
-    # catch intermittent failures
-    set -eo pipefail
+     call ValidateMatrix as ValidateMatrix {
+         input:
+             matrix = matrix,
+             matrix_row_index = matrix_row_index,
+	     matrix_col_index = matrix_col_index,
+	     reference_matrix = reference_matrix
+     }
+
+     call ValidateMetrics {
+         input:
+             cell_metrics = cell_metrics,
+             gene_metrics = gene_metrics,
+             expected_cell_metric_hash = expected_cell_metric_hash,
+             expected_gene_metric_hash = expected_gene_metric_hash
+     }
 
-    # calculate hashes; awk is used to extract the hash from the md5sum output that contains both
-    # a hash and the filename that was passed. gzipped files are unzipped to avoid hashing malleable
-    # metadata
+     call GenerateReport as GenerateReport {
+         input:
+             bam_validation_result = ValidateBam.result,
+             matrix_validation_result = ValidateMatrix.result,
+             metric_and_index_validation_result = ValidateMetrics.result
+    }
+}
 
+task ValidateBam {
+    File bam
+    String expected_checksum
+    Int required_disk = ceil(size(bam, "G") * 1.1)
 
-    unzip "${matrix}"
-    matrix_hash=$(find . -name "*.npy" -type f -exec md5sum {} \; | sort -k 2 | md5sum | awk '{print $1}')
-    gene_metric_hash=$(zcat "${gene_metrics}" | md5sum | awk '{print $1}')
-    cell_metric_hash=$(zcat "${cell_metrics}" | md5sum | awk '{print $1}')
+    command <<<
+        echo Starting checksum generation...
+        # calculate hash for alignment positions only (a reduced bam hash)
+        samtools view -F 256 "${bam}" | cut -f 1-11 | md5sum | awk '{print $1}' > md5_checksum_reduced.txt
+        echo Reduced checksum generation complete
 
-    # calculate hash as above, but ignore run-specific bam headers
-    bam_hash=$(samtools view "${bam}" | md5sum | awk '{print $1}')
+        calculated_checksum=$(cat md5_checksum_reduced.txt)
```

Comment:
You could avoid the file and write directly to the environment variable.
```
calculated_checksum=$(samtools view -F ... | awk '{print $1}')
```

Reply:
done